app/src/main/java/org/fossasia/pslab/communication/sensors/AD7718.java

```diff
@@ -0,0 +1,326 @@
+        int stat = readRegister(STATUS);
+        String[] P = {"PLL LOCKED", "RES", "RES", "ADC ERROR", "RES", "CAL DONE", "RES", "READY"};
+        String[] N = {"PLL ERROR", "RES", "RES", "ADC OKAY", "RES", "CAL LOW", "RES", "NOT READY"};
+        StringBuilder sb = new StringBuilder();
+        for (int a = 0; a < 8; a++) {
+            if ((stat & (1 << a)) == 1) {
+                sb.append(P[a]);
+            } else {
+                sb.append(N[a]);
+            }
+        }
+        Log.d(TAG, stat + ", " + sb.toString());
+    }
+
+    public float convert_unipolar(float x) {
+        return (float) (1.024 * VREF * x) / (gain * 2 ^ 24);
+    }
+
+    public float convert_bipolar(float x) {
+        return (float) (((x / (2 ^ 24)) - 1) * (1.024 * VREF) / (gain));
+    }
+
+    private boolean startRead(String chan) throws IOException {
+        List channels = Arrays.asList(CHAN_NAMES);
+        if (channels.contains(chan)) {
+            int channelID = channels.indexOf(chan);
+            configADC(CON_RANGE7 | CON_UNIPOLAR | channelID << 4);
+            writeRegister(MODE, MODE_SINGLE | MODE_CHCON | MODE_REFSEL);
+            return true;
+        } else  {
+            Log.d(TAG, "Invalid Channel Name. try AIN1AINCOM");
+            return false;
+        }
+    }
+
+    private boolean fetchData(String chan) throws IOException, InterruptedException {
+
+        while (true) {
+            int stat = readRegister(STATUS);
+            if ((stat & 0x80) == 1) {
+                float data = readData();
+                data = convert_unipolar(data);
+                List channelList = Collections.singletonList(chan);
+                if ((int) channelList.get(3) > 4) {
+                    data = (data - 3.3f / 2) * 4;
+                }
+                PolynomialFunction function = new PolynomialFunction(caldata.get(chan));
+                return function.value(data) == 0;
+            } else {
+                Thread.sleep(100);
+                Log.d(TAG, "Increase Delay");
+            }
+        }
+    }
+
+    public boolean readVoltage(String channel) throws IOException, InterruptedException {
+        if (startRead(channel)) {
+            return false;
+        }
+        Thread.sleep(150);
+        return fetchData(channel);
+    }
+
+    private boolean fetchRawData(String chan) throws IOException, InterruptedException {
+        while (true) {
+            int stat = readRegister(STATUS);
+            if ((stat & 0x80) == 1) {
+                float data = readData();
+                return convert_unipolar(data) == 1;
+            } else {
+                Thread.sleep(100);
+                Log.d(TAG, "Increase Delay");
+            }
+        }
+    }
+
+    public boolean readRawVoltage(String channel) throws IOException, InterruptedException {
+        if (startRead(channel)) {
+            return false;
+        }
+        Thread.sleep(150);
+        return fetchRawData(channel);
+    }
+
```

Comment:
And give a meaning full name wherever you can Please. "bv" doesn't give a clue to what it does.